src/couch_server.erl

```diff
@@ -116,12 +116,16 @@ delete(DbName, Options) ->
 maybe_add_sys_db_callbacks(DbName, Options) when is_binary(DbName) ->
     maybe_add_sys_db_callbacks(?b2l(DbName), Options);
 maybe_add_sys_db_callbacks(DbName, Options) ->
+    Normalized = couch_db:normalize_dbname(DbName),
+    IsShard = couch_db:is_shard(DbName),
+
     DbsDbName = config:get("mem3", "shards_db", "_dbs"),
     NodesDbName = config:get("mem3", "nodes_db", "_nodes"),
-    IsReplicatorDb = DbName == config:get("replicator", "db", "_replicator") orelse
-	path_ends_with(DbName, <<"_replicator">>),
-    IsUsersDb = DbName ==config:get("couch_httpd_auth", "authentication_db", "_users") orelse
-	path_ends_with(DbName, <<"_users">>),
+    IsReplicatorDb = Normalized == config:get("replicator", "db", "_replicator") orelse
```

Comment:
As I read this, Normalized is guaranteed to be a binary, where config:get/3 is guaranteed to be a list, so the first part of the orelse is guaranteed to be false.